### DISCUSSION TITLE
Trakt Integration with Shows List

### DISCUSSION THREAD

**Discussion by knowair** [pymedusa/Medusa#6426] [Ideas & Feature requests]
**Is your feature request related to a problem? Please describe.**
Enhancement to the current active Trakt integration already in place

**Describe the solution you'd like**
Currently you can link Trakt to push newly downloaded shows up to add to your Collection.

I'd like to be able to do the reverse(?) and pull from my Trakt shows in Progress and add them to my Medusa Shows list automatically. 

This would ideally be One Click to pull the list down from Trakt and the shows added to Medusa based on default settings. 

Once the first sync is done each subsequent sync would then check Trakt for missing shows to add and upload any newly download to the collection as now. 

**Describe alternatives you've considered**
It could also come in the format of viewing your Trakt Progress in the same way you view the Trakt "Most Popular"  i.e. once you connect your Trakt account have "My Trakt Shows" in the "Add Shows" sections for adding manually. 

**Additional context**
This would create a "Full circle" of automation where new shows added as collected or watched on Trakt would automatically be populated into Medusa - then as they are downloaded the new episodes are added back to the Trakt collection. 

A user watches a random show on live TV in Kodi/Plex. that show is marked via the Kodi>Trakt scrobbling and Medusa syncs to start downloading it without the user having to do a thing!

It acts as a backup for users too, if a user loses their setup, all they need to do is connect to trakt and the list of active and collected shows can be populated to Medusa and Medusa can start downloading the missing and next episodes.

**Comment by Thraxis** [comment]
Unless something has changed this should already be the behavior of the Trakt "Sync Library" and "Sync Watchlist" functions.

IE adding a show to your trakt watchlist / library should add the show Medusa if those two options are enabled.

**Comment by knowair** [comment]
Sorry if that's the case, I couldn't get the library sync to work 2 ways like this. It only seems to notify Trakt of new downloads by updating the Trakt collection. Not populating the show list automatically. 

I hadn't tried the watchlist sync because I didn't want the Watchlist shows downloaded, they are the shows that I Might watch, so have saved there for the future, when I have a gap in my schedule!! So many great shows not enough time!!

**Comment by knowair** [comment]
OK, As an Example for this.

The new season of Line of duty has started a couple of weeks ago, I have this show marked as Watched and Collected (dvd) on Trakt but I did not have any of these on my local system so it had not been added via the "Add existing shows from folders" function.

It would be great if as part of the the Trakt Sync function there was an option that checked for shows that are listed on Trakt as watched and/or collected and added them to "My Shows" automatically. 

Maybe it could be set to check for upcoming shows via the user's Trakt calendar, only adding shows that are missing locally when a new episode appears on the calendar. 

Maybe have an option in the settings to have them added in a paused state so that the user can .

**Comment by RobLoach** [comment]
Another potential use of Trakt would be to take advantage of Medusa's Episode Status. For each "Watched" episode in Trakt......

- Episodes in Medusa marked as "Wanted" become "Skipped"
- Episodes in Medusa marked as "Snatched" become "Skipped"
- Episodes in Medusa marked as "Failed" become "Skipped"
- Episodes in Medusa marked as "Downloaded" become "Archived"
